python/pyiceberg/catalog/__init__.py

```diff
@@ -53,7 +54,15 @@ class Catalog(ABC):
 
     def __init__(self, name: str | None, **properties: str):
         self.name = name
-        self.properties = properties
+
+        if name is not None:
```

Comment:
Should name be required instead? When does it make sense to create a catalog with no name?

Reply:
In the first instance, I would a name to be mandatory. I changed this because of the configuration. If you don't have a name, it won't load the config environment variable `PYICEBERG_CATALOG__DEFAULT__URI`, and fall back to `PYICEBERG_URI` in the CLI. I can imagine that we want to change this, but I'm not fully settled on this one either.

What I don't really like is that the `PYICEBERG_URI` and `PYICEBERG_CREDENTIAL` are REST specific, and I wouldn't like to open a door to have another way of setting config for other catalog implementation in the future (for Glue PYICEBERG_ACCOUNT_ID for example).

Reply:
URI isn't specific to REST. JDBC and Hive also use URI. Credential is REST-specific though.

For the environment variables, we can always remove them and add them back when we find a good way to manage them. We just need to deconflict the existing ones (`PYICEBERG_URI`, etc.) and the new-style ones `PYICEBERG__CATALOG__NAME__URI` right?

I think we have two options:
1. `PYICEBERG_URI` configures an anonymous catalog with a random name that can only be used if you don't specify `--catalog`
2. Remove `PYICEBERG_URI` and only support `PYICEBERG__*` properties

I don't really mind either way. If we're going to support the `PYICEBERG__*` properties in general then it probably makes sense to get rid of the existing ones.

Reply:
GlueCatalog wouldn't have a URI, but let's not overcomplicate things. I made it mandatory.

I also removed `PYICEBERG_URI` I think it is better and less confusing in the long run to have on the way of setting the environment variables. I also refactored the properties, mostly by changing all the `__init__` methods of the catalogs:
```python
    def __init__(
        self,
        name: str,
        **properties: str,
    ):
```
This way we don't have any specifics in the Catalog itself.